software/openvisualizer/openvisualizer/remoteConnectorServer/remoteConnectorServer.py

```diff
@@ -1,109 +1,223 @@
+        """
+        context = zmq.Context()
+        admin = context.socket(zmq.REP)
+        admin.setsockopt(zmq.IPV6, 1)
+        admin.setsockopt(zmq.IPV4ONLY, 0)
+        admin.bind("tcp://*:%s" % self.zmq_admin_port)
+
+        def _handle_pub_all_callback(**kwargs):
+            # Assume here that we got a JSON
+            self.queue.put(kwargs)
+
+        subscriptions = set()
         while True:
-            event = self.subscriber.recv_json()
+            #  Wait for next request from client
+            message = admin.recv()
+            log.info("Received request: %s" % message)
+
+            try:
+                message = json.loads(message)
+            except Exception as e:
+                print("That was not a JSON")
+                log.error(e)
+                continue
+
+            if not set(message.keys()).issubset({"sub", "unsub"}):
+                admin.send_json({"error": "Incorrect format use a JSON with sub, unsub"})
+                continue
+            count = 0
+            if "sub" in message.keys():
+                for sub in message["sub"]:
+                    if sub not in subscriptions:
+                        dispatcher.connect(_handle_pub_all_callback, signal=sub)
+                        count += 1
+                        subscriptions.add(sub)
+            if "unsub" in message.keys():
+                for unsub in message["unsub"]:
+                    if unsub in subscriptions:
+                        dispatcher.disconnect(_handle_pub_all_callback, signal=unsub)
+                        count += 1
+                        subscriptions.remove(unsub)
+            admin.send_json({"subscriptions": list(subscriptions)})
+
+    def _handle_sub(self):
+        """
+
+        :return:
+        """
+        context = zmq.Context()
+        subscriber = context.socket(zmq.SUB)
+        subscriber.setsockopt(zmq.IPV6, 1)
+        subscriber.setsockopt(zmq.IPV4ONLY, 0)
+        subscriber.setsockopt(zmq.SUBSCRIBE, "")
+        log.info("Subscriber started")
+
+        count = 0
+        while True:
+            event = subscriber.recv_json()
             if count > 10:
-                log.info("Received remote event\n"+json.dumps(event)+"\nDispatching to event bus")
+                log.info("Received remote event\n" + json.dumps(event) + "\nDispatching to event bus")
                 count = 0
             dispatcher.send(
-                sender  =  event['sender'].encode("utf8"),
-                signal  =  event['signal'].encode("utf8"),
-                data    =  event['data']
+                sender=event['sender'].encode("utf8"),
+                signal=event['signal'].encode("utf8"),
+                data=event['data']
+            )
+            count += 1
+
+    def _handle_req(self):
+        """
+        Function call when external event are injected in the message bus
+        :return:
+        """
+        context = zmq.Context()
+        rep_socket = context.socket(zmq.REP)
```

Comment:
@sieben I guess you may trying to use PULL here rather than REP, as the tutorial said there is a PULL socket for handling sending message. https://openwsn.atlassian.net/wiki/pages/viewpage.action?pageId=113475628#Sending/ReceivingmessagesontheEventBususingZMQ-Sendingmessages